lib/passport-saml/saml.js

```diff
@@ -392,29 +392,44 @@ SAML.prototype.getAuthorizeForm = function (req, callback) {
       samlMessage[k] = additionalParameters[k] || '';
     });
 
-    var formInputs = Object.keys(samlMessage).map(function(k) {
-      return '<input type="hidden" name="' + k + '" value="' + quoteattr(samlMessage[k]) + '" />';
-    }).join('\r\n');
-
-    callback(null, [
-      '<!DOCTYPE html>',
-      '<html>',
-      '<head>',
-      '<meta charset="utf-8">',
-      '<meta http-equiv="x-ua-compatible" content="ie=edge">',
-      '</head>',
-      '<body onload="document.forms[0].submit()">',
-      '<noscript>',
-      '<p><strong>Note:</strong> Since your browser does not support JavaScript, you must press the button below once to proceed.</p>',
-      '</noscript>',
-      '<form method="post" action="' + encodeURI(self.options.entryPoint) + '">',
-      formInputs,
-      '<input type="submit" value="Submit" />',
-      '</form>',
-      '<script>document.forms[0].style.display="none";</script>', // Hide the form if JavaScript is enabled
-      '</body>',
-      '</html>'
-    ].join('\r\n'));
+    function toEscapedNameValue(key) {
+      return {
+        name: key,
+        value: quoteattr(samlMessage[key])
+      };
+    }
+
+    var escapedNameValues = Object.keys(samlMessage).map(toEscapedNameValue);
+
+    function createDefaultHtmlForm() {
+      var formInputs = escapedNameValues.map(function (input) {
+        return '<input type="hidden" name="' + input.name + '" value="' + input.value + '" />';
+      }).join('\r\n');
+
+      return [
+        '<!DOCTYPE html>',
+        '<html>',
+        '<head>',
+        '<meta charset="utf-8">',
+        '<meta http-equiv="x-ua-compatible" content="ie=edge">',
+        '</head>',
+        '<body onload="document.forms[0].submit()">',
+        '<noscript>',
+        '<p><strong>Note:</strong> Since your browser does not support JavaScript, you must press the button below once to proceed.</p>',
+        '</noscript>',
+        '<form method="post" action="' + encodeURI(self.options.entryPoint) + '">',
+        formInputs,
+        '<input type="submit" value="Submit" />',
+        '</form>',
+        '<script>document.forms[0].style.display="none";</script>', // Hide the form if JavaScript is enabled
+        '</body>',
+        '</html>'
+      ].join('\r\n');
+    }
+
+    var postBindingForm = createCustomForm ? createCustomForm(escapedNameValues) : createDefaultHtmlForm();
```

Comment:
Would suggest a signature for createCustomForm as: `createCustomForm(escapedNameValues, entryPoint, callback)` so that the entryPoint from options will be available as well, and allow the function to be potentially asynchronous. passed entryPoint value should already be encoded (as on line 420)